addons/mail/models/res_users.py

```diff
@@ -137,6 +137,28 @@ def _get_portal_access_update_body(self, access_granted):
             return '%s (%s)' % (body, self.partner_id.email)
         return body
 
+    @api.model
+    def _deactivate_portal_user(self, **post):
+        """Blacklist the email of the user after deleting it.
+
+        Log a note on the related partner so we know why it's archived.
+        """
+        for user in self:
+            user.partner_id._message_log(body=_(
+                'This partner was archived because the related user "%s" has requested to deactivate his account.',
```

Comment:
Maybe ``This partner has been archived because %(user_name)s (#%(user_id)s) asked to deactivate this account`` ?